spec.emu

```diff
@@ -502,28 +690,28 @@ contributors: Luca Casonato, Guy Bedford, Nicolò Ribaudo
 
             <emu-alg>
               1. Assert: _state_.[[IsLoading]] is *true*.
-              1. <ins>Let _importOnlyAsPhase_ be *false*.</ins>
-              1. <ins>If _referrer_ has been specified, then</ins>
-                1. <ins>Assert: _specifier_ has been specified.</ins>
-                1. <ins>If _referrer_.[[RequestedSourcePhasesOnly]] contains _specifier_, set _importOnlyAsPhase_ to *true*.</ins>
-              1. If <ins>_importOnlyAsPhase_ is *false*,</ins> _module_ is a Cyclic Module Record, _module_.[[Status]] is ~new~, and _state_.[[Visited]] does not contain _module_, then
+              1. If _module_ is a Cyclic Module Record, _module_.[[Status]] is ~new~, and _state_.[[Visited]] does not contain _module_, <ins>and _phase_ is ~evaluation~, </ins>then
                 1. Append _module_ to _state_.[[Visited]].
-                1. Let _requestedModulesCount_ be the length of _module_.[[RequestedModules]].
-                1. Set _state_.[[PendingModulesCount]] to _state_.[[PendingModulesCount]] + _requestedModulesCount_.
-                1. For each String _required_ of _module_.[[RequestedModules]], do
-                  1. If _module_.[[LoadedModules]] contains a Record _record_ whose [[Specifier]] is _required_, then
-                    1. Perform InnerModuleLoading(_state_, _record_.[[Module]], <ins>_module_, _required_</ins>).
+                1. <del>Let _requestedModulesCount_ be the number of elements in _module_.[[RequestedModules]].</del>
+                1. <del>Set _state_.[[PendingModulesCount]] to _state_.[[PendingModulesCount]] + _requestedModulesCount_.</del>
+                1. <del>For each String _required_ of _module_.[[RequestedModules]], do</del>
+                1. <ins>For each ModuleRequest Record _required_ of _module_.[[RequestedModules]], do</ins>
+                  1. If _module_.[[LoadedModules]] contains a Record whose [[Specifier]] is _required_<ins>.[[Specifier]]</ins>, then
```

Comment:
This PR now duplicates entries in the `[[ReqeustedModules]]` list in case there are different phases (i.e. `import source s from "x"; import x from "x"`), however this check still skips calling `HostLoadImporteedModule` is that module has already been loaded even if for a different phase. Is this intentional?